Add possibility to do auth for Aqua SaaS near default auth Aqua.

Ukhanskyi opened the pull request:
Trying to connect to Aqua SaaS, near existing authentication logic.

[GitHub Issue](https://github.com/KennaSecurity/toolkit/issues/550#issue-2362548416)

Ukhanskyi commented:
@jgarber, @shotop, @scotttesler, @JaiHirsch, Hi everyone 👋🏼)
Would it be possible for someone to review these changes for the Aqua connector and merge them if everything is correct?

Ukhanskyi commented:
@jgarber-cisco, or maybe you could to check this code during your business hours.

jgarber-cisco commented on tasks/connectors/aqua/lib/aqua_helper.rb:
```diff
@@ -5,30 +5,77 @@
-
-        @headers = { "Content-Type" => "application/json" }
-        payload = {
-          "id": username.to_s,
```
I don't have access to the API docs to know: is `email` on line 31 an acceptable substitute for `id` when using Aqua on-prem?

Ukhanskyi replied:
Me too, so I've reverted id when using Aqua on-prem

jgarber-cisco reviewed (approved):
I have a question about whether this breaks on-prem, but otherwise, LGTM!

jgarber-cisco commented on tasks/connectors/aqua/lib/aqua_helper.rb:
```diff
@@ -5,30 +5,77 @@
-          "id": username.to_s,
-          "password": password.to_s
-        }
+        if aqua_url.include?("cloud.aquasec.com")
```
Oh, that's a good suggestion! +1

jgarber-cisco commented on tasks/connectors/aqua/aqua.rb:
```diff
@@ -102,6 +103,8 @@
         token = aqua_get_token(aqua_url, username, password)
         fail_task "Unable to authenticate with Aqua, please check credentials" unless token
 
+        aqua_url = get_wp_url(token) if aqua_url.include?("cloud.aquasec.com")
```
Probably you'll want to abstract the two checks for `cloud.aquasec.com` into a method as your url check gets a little more sophisticated.

jgarber-cisco commented:
@Ukhanskyi Rebase to pick up the security upgrades from `main`

jgarber-cisco reviewed (changes requested):
Rebase, look at on-prem, and make cloud url checking safer.

Ukhanskyi commented:
> Rebase, look at on-prem, and make cloud url checking safer.

Rebased on the main branch and squashed all commits. I returned the 'id' for on-prem, and also tried to make the URL check more accurate and safe. Could you review this code one more time, please

jgarber-cisco reviewed (approved):
Great job!
